dlt/normalize/normalize.py

```diff
@@ -110,33 +110,70 @@ def w_normalize_files(
     ) -> TWorkerRV:
         destination_caps = config.destination_capabilities
         schema_updates: List[TSchemaUpdate] = []
-        item_normalizers: Dict[TDataItemFormat, ItemsNormalizer] = {}
+        item_normalizers: Dict[Tuple[TDataItemFormat, str], ItemsNormalizer] = {}
         # Use default storage if parquet is not supported to make normalizer fallback to read rows from the file
         preferred_file_format = (
             destination_caps.preferred_loader_file_format
             or destination_caps.preferred_staging_file_format
         )
         # TODO: capabilities.supported_*_formats can be None, it should have defaults
-        supported_formats = destination_caps.supported_loader_file_formats or []
+        supported_file_formats = destination_caps.supported_loader_file_formats or []
+        supported_table_formats = destination_caps.supported_table_formats or []
 
         # process all files with data items and write to buffered item storage
         with Container().injectable_context(destination_caps):
             schema = Schema.from_stored_schema(stored_schema)
             normalize_storage = NormalizeStorage(False, normalize_storage_config)
-            load_storage = LoadStorage(False, supported_formats, loader_storage_config)
+            load_storage = LoadStorage(False, supported_file_formats, loader_storage_config)
+
+            def _get_items_normalizer(
+                item_format: TDataItemFormat, table_schema: TTableSchema
+            ) -> ItemsNormalizer:
+                if (item_format, table_schema["name"]) in item_normalizers:
+                    return item_normalizers[(item_format, table_schema["name"])]
+
+                if (
+                    "table_format" in table_schema
+                    and table_schema["table_format"] not in supported_table_formats
+                ):
+                    logger.warning(
+                        "Destination does not support the configured `table_format` value "
+                        f"`{table_schema['table_format']}` for table `{table_schema['name']}`. "
+                        "The setting will probably be ignored."
+                    )
+
+                items_preferred_file_format = preferred_file_format
+                items_supported_file_formats = supported_file_formats
+                if destination_caps.loader_file_format_adapter is not None:
+                    items_preferred_file_format, items_supported_file_formats = (
+                        destination_caps.loader_file_format_adapter(
+                            preferred_file_format,
+                            supported_file_formats,
```

Comment:
maybe we should copy this list here... just in case someone modifies input list in place